Editor: More options: Add info bubble to pingbacks and trackbacks

arunsathiya opened the pull request:
#### Changes proposed in this Pull Request

* Add info bubble to pingbacks and trackbacks section on Calypso editor
* Links the keywords to the respective support documents

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Launch the live branch link available below
* Click on `Add` next to `Blog posts`/`Site pages` 
* Click `More options` section on the right
* Under `Discussion`, find `Allow Pingbacks & Trackbacks`
* Click on the info bubble available next to it
* Ensure the bubble looks aligned, has the appropriate text mentioned in #22679, and that it links to the support documents

#### Before

<img width="700" alt="screenshot 2018-12-21 at 18 18 12" src="https://user-images.githubusercontent.com/18581859/50343516-30bfaf00-054d-11e9-98dd-921748c6312b.png">

#### After

<img width="700" alt="screenshot 2018-12-21 at 18 19 03" src="https://user-images.githubusercontent.com/18581859/50343509-27364700-054d-11e9-92e1-0f0819b4d4f7.png">

<img width="700" alt="screenshot 2018-12-21 at 18 17 15" src="https://user-images.githubusercontent.com/18581859/50343512-2ac9ce00-054d-11e9-9c1a-b65cad73c2da.png">

Fixes #22679

arunsathiya commented:
I am following the CSS namespace guideline for the new info bubble, similar to how it was done for the previous, comment status' info bubble. Wonder how I can fix this. 🤔

tyxla commented on client/post-editor/editor-discussion/index.jsx:
```diff
@@ -136,7 +136,37 @@ export class EditorDiscussion extends React.Component {
+						{ this.props.translate( 'Allow Pingbacks & Trackbacks' ) }
+						<InfoPopover
+							position="top right"
+							className="editor-ping_status__info"
```
This classname, and the pre-existing `editor-comment_status__info` seem to not be following our [coding standards](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md#classes). While I realize the new field is only following the pattern of the above field, it's a great reason to fix it for the above case too.

The reason why `editor-comment_status__info` is not valid is because in classnames:
* We should use only one component prefix - in our case it should be `editor-discussion`, otherwise the linter will complain.
* Component prefix should be separated from the element by a double underscore (`__`).
* Both the component prefixes and elements can contain only dashes (`-`) but not underscores (`_`).

As you'll see in a comment below, I suggest that we change `.editor-comment_status__info` to `.editor-discussion__status-info`, and use it for both the info popovers.

Let me know if you have any questions though!

arunsathiya replied:
That's a great explanation. Thanks, Marin!

arunsathiya commented on client/post-editor/editor-discussion/style.scss:
```diff
@@ -6,3 +6,12 @@
 	}
 }
+
+.editor-ping_status__info {
```
That's a good idea as well. I have addressed this in my update on 490351f.

arunsathiya commented on client/post-editor/editor-discussion/index.jsx:
```diff
@@ -136,7 +136,37 @@ export class EditorDiscussion extends React.Component {
+								{
+									components: {
+										a: (
+											<a
```
That's a very good idea as well. I have addressed that in c096d8a.

arunsathiya commented:
Thanks so much for taking your time to review these, @tyxla. 

Hopefully I have addressed all your suggestions, and if something's off, please let me know and I am happy to address that as well!

arunsathiya commented on client/post-editor/editor-discussion/index.jsx:
```diff
@@ -136,7 +137,37 @@ export class EditorDiscussion extends React.Component {
+							popoverName="PingStatus"
+						>
+							{ this.props.translate(
+								'{{pingbacksLink}}Pingbacks{{/pingbacksLink}} and {{trackbacksLink}}trackbacks{{/trackbacksLink}} are automated comments you will receive when others create links to your post elsewhere.',
```
That is another great tip, thanks! I keep learning from you. I have addressed that in 6997e8b.

arunsathiya commented on client/post-editor/editor-discussion/index.jsx:
```diff
@@ -136,7 +137,37 @@ export class EditorDiscussion extends React.Component {
+											<ExternalLink
+												href="https://support.wordpress.com/comments/trackbacks/"
+												target="_blank"
+												icon="true"
```
Excellent, the images in the original post have been updated now. ✅ 

I am leaving this conversation open, in case any design members want to chime in on the same thread.

folletto commented:
I gave a quick review. I wasn't unfortunately able to boot from live (maybe it needs a rebase?) but from the code and the screenshots:

1. Visually you followed the existing pattern, and it doesn't clashes with anything I can see, so looks good.
2. I'm not sure about the class name `editor-discussion__status-info`, and I'd change it to `editor-discussion__info-bubble` while @tyxla is correct in fixing the linter issue above, "status-info" isn't descriptive of anything as the description was meant to be "editor discussion status", and "status info" doesn't describe... anything? So let's keep the component name `editor-discussion` and append a meaningful name, `info-bubble`.

arunsathiya commented:
That's a great point, @folletto! I have gone ahead and made this change on 0f7a204 ✅
